core/src/scene/scene.cpp

```diff
@@ -82,9 +83,25 @@ bool Scene::texture(const std::string& textureName, std::shared_ptr<Texture>& te
     return true;
 }
 
-void Scene::queueComponentUpdate(std::string componentPath, std::string value) {
-    std::vector<std::string> splitPath = splitString(componentPath, COMPONENT_PATH_DELIMITER);
-    m_updates.push_back({ splitPath, value });
+void Scene::queueUpdate(std::string path, std::string value) {
+    auto keys = splitString(path, COMPONENT_PATH_DELIMITER);
+    m_updates.push_back({ keys, value });
+}
+
+void Scene::addClientDataSource(std::shared_ptr<DataSource> _source) {
+    m_clientDataSources.push_back(_source);
+}
+
+void Scene::removeClientDataSource(DataSource& _source) {
+    auto it = std::remove_if(m_clientDataSources.begin(), m_clientDataSources.end(),
+        [&](auto& s) { return s.get() == &_source; });
+    m_clientDataSources.erase(it, m_clientDataSources.end());
```

Comment:
I do not get this second `erase` here.

First `remove_if` will remove all the clients datasources and return `past the end iterator` for the set of elements remaining in the container.
And the second statement is erasing everything from the `past the end iterator` (which is/should be `m_clientDataSources.end()`) to `m_clientDataSources.end()`. Not sure if the second erase is needed here.

Can you explain if I am misreading something here?

Thanks